presc/evaluations/conditional_distribution.py

```diff
@@ -0,0 +1,189 @@
+    """
+
+    def __init__(self, vals, bins, categorical, config=None):
+        self.vals = vals
+        self.bins = bins
+        self.categorical = categorical
+
+    def display_result(self, xlab, config=None):
+        """Display the distributions for the given data column.
+
+        xlab: label to display on the x-axis
+        ylab: label to display on the y-axis
+        config: optional dict of config options to override the defaults
+        """
+        _config = dict(DEFAULT_CONFIG)
+        if config:
+            _config.update(config)
+
+        for y_true, y_pred in self.vals.index.droplevel(-1).unique():
+            counts = self.vals.loc[(y_true, y_pred)]
+            if isinstance(self.bins.index, MultiIndex):
+                bins = self.bins.loc[(y_true, y_pred)]
+            else:
+                bins = self.bins
+            if self.categorical:
+                plt.bar(
+                    bins.astype("str"),
+                    counts,
+                )
+            else:
+                bins = bins.values
+                plt.hist(
+                    (bins[:-1] + bins[1:]) / 2,
+                    bins=len(counts),
+                    weights=counts,
+                    range=(bins.min(), bins.max()),
+                )
+            plt.xlabel(xlab)
+            plt.ylabel("Frequency")
+            plt.title(f"Group: {y_true}_predicted_as_{y_pred}")
+
+            plt.show(block=False)
+
+
+class ConditionalDistribution:
+    """Computation of data distributions conditional on prediction results.
+
+    model: the ClassificationModel to run the evaluation for
+    test_dataset: a Dataset to use for evaluation.
+    config: optional dict of config options to override the defaults
+    """
+
+    def __init__(self, model, test_dataset, config=None):
+        _config = dict(DEFAULT_CONFIG)
+        if config:
+            _config.update(config)
+        self._config = _config
+        self._model = model
+        self._test_dataset = test_dataset
+        self._test_pred = self._model.predict_labels(test_dataset)
+
+    def compute_for_column(self, colname, as_categorical=False):
+        """Compute the evaluation for the given dataset column.
+
+        colname: a column in the dataset to compute distributions for
+        as_categorical: should the feature be treated as categorical, ie.
+            binned on its unique values? If the feature is not numeric, this
+            param is ignored.
+        Returns a `ConditionalDistributionResult` instance.
+        """
+        return compute_conditional_distribution(
+            data_col=self._test_dataset.df[colname],
+            true_labs=self._test_dataset.labels,
+            pred_labs=self._test_pred,
+            as_categorical=as_categorical,
+            config=self._config,
+        )
+
+    def display(self, colnames=None):
+        """Displays the conditional distribution result for each specified column.
+
```

Comment:
Indicate that it computes and displays so there is not confusion about another step involved